linera-chain/src/chain.rs

```diff
@@ -393,7 +393,7 @@ where
             let OutgoingMessage {
                 destination,
                 authenticated_signer,
-                is_skippable,
+                is_protected,
```

Comment:
Do we need that field, if it depends only on the message variant?

Reply:
Only system messages that never fails can be "protected" (counter-example: `Withdraw` which now correctly fails after ).

The notion of "protected" messages grew on me after I realized that:
* It helps with error handling during block construction (until we have something better).
* Together with fallback leaders, this will ensure that messages are eventually executed (within a reasonable timeframe). ("mandatory messages" require messages to be accepted in the next block but the implementation is more complex. We may not even need them in the end.)

Reply:
(Yes, I'm just saying it might not have to be a field in `OutgoingMessage`; it could just be a function, computed from the message variant.)

Reply:
But we want the knowledge about system commands to be localized un system.rs right?

Reply:
Makes sense.
I guess `SystemMessage` could have an `is_protected` method (so the user doesn't need to know about the variants); but I'm fine with the field, too, of course!